Assets/Scripts/Agents/AgentBase.cs

```diff
@@ -256,109 +260,60 @@ private void AlignWithVelocity() {
+    float horizontal = Mathf.Sqrt(horizontalSqr);
+    float rangeSqr = horizontalSqr + y * y;
+    float range = Mathf.Sqrt(rangeSqr);
+
+    float azimuth = Mathf.Atan2(x, z);
+    float elevation = Mathf.Atan2(y, horizontal);
+    var positionTransformation = new PositionTransformation {
       Cartesian = relativePosition,
-      Range = relativePosition.magnitude,
+      Range = range,
       Azimuth = azimuth,
       Elevation = elevation,
     };
-  }
-
-  private VelocityTransformation GetRelativeVelocityTransformation(in Vector3 relativePosition,
-                                                                   in Vector3 relativeVelocity) {
-    if (relativePosition.sqrMagnitude < _epsilon) {
-      return new VelocityTransformation {
-        Cartesian = relativeVelocity,
-        Range = relativeVelocity.magnitude,
-        Azimuth = 0,
-        Elevation = 0,
-      };
-    }
 
-    // Calculate range rate (radial velocity).
-    float rangeRate = Vector3.Dot(relativeVelocity, relativePosition.normalized);
-
-    // Project relative velocity onto the sphere passing through the target.
-    Vector3 tangentialVelocity = Vector3.ProjectOnPlane(relativeVelocity, relativePosition);
-
-    // The target azimuth vector is orthogonal to the relative position vector and points to the
-    // starboard of the target along the azimuth-elevation sphere.
-    Vector3 targetAzimuth = Vector3.Cross(Up, relativePosition);
-    // The target elevation vector is orthogonal to the relative position vector and points upward
-    // from the target along the azimuth-elevation sphere.
-    Vector3 targetElevation = Vector3.Cross(relativePosition, Right);
-    // If the relative position vector is parallel to the yaw or pitch axis, the target azimuth
-    // vector or the target elevation vector will be undefined.
-    if (targetAzimuth.sqrMagnitude < _epsilon) {
-      targetAzimuth = Vector3.Cross(targetElevation, relativePosition);
-    } else if (targetElevation.sqrMagnitude < _epsilon) {
-      targetElevation = Vector3.Cross(relativePosition, targetAzimuth);
+    float rangeRate =
+        range > _epsilon ? Vector3.Dot(relativeLocalVelocity, relativeLocalPosition) / range : 0f;
+    float azimuthRate = 0f;
+    float elevationRate = 0f;
+    if (horizontal > _epsilon) {
+      azimuthRate = -(x * relativeLocalVelocity.z - z * relativeLocalVelocity.x) / horizontalSqr;
+      elevationRate =
+          (relativeLocalVelocity.y * horizontal -
+           y * (x * relativeLocalVelocity.x + z * relativeLocalVelocity.z) / horizontal) /
+          rangeSqr;
+    } else {
+      // The other agent is exactly above or below.
+      azimuthRate = 0f;
+      float horizontalSpeed = Mathf.Sqrt(relativeLocalVelocity.x * relativeLocalVelocity.x +
+                                         relativeLocalVelocity.z * relativeLocalVelocity.z);
+      elevationRate = -horizontalSpeed / (Mathf.Abs(y) > _epsilon ? y : Mathf.Sign(y) * _epsilon);
```

Comment:
Potential divide-by-zero here since Mathf.Sign(0) returns 0 so if y is 0 (exactly at same alt), this becomes div-by-zero.